### DISCUSSION TITLE
Vite push scripts and styles

### DISCUSSION THREAD

**Discussion by remotemerge** [laravel/framework#43194] [Q&A] [answered]
I used to push the scripts and styles from the inner pages. It was possible with Webpack and Laravel Mix.

```
@push('scripts')
  <script type="text/javascript" src="{{ asset('/js/swiper.js') }}"></script>
@endpush
```

Now I am testing Vite. Unfortunately, I can't use `push` like above. The swiper is required on the gallery page and not on the other 100s of pages.
```
@vite('resources/js/swiper.js')
```
One way I found is to parse `manifest.json` read the path of hashed script and style and push it manually. It works :) We need some workaround on this, or maybe it's already there; I don't know!

**Comment by iksaku** [comment, marked as answer]
I can actually push vite resources without issues. Just wrap the `@vite` directive in `@push()...@endpush` directives

```blade
{{-- Works --}}
@push('scripts')
    @vite('resources/js/app.js')
@endpush

{{-- This works as well --}}
@pushonce('scripts')
    @vite('resources/js/app.js')
@endpushonce
```
